### PR TITLE
Greetz

### PR DESCRIPTION
**Pull request by RipaAloyan**

### CODE REVIEW

File: WebsiteGreetz/src/test/java/WebsiteTest.java [hunk 1]
```diff
@@ -24,56 +22,53 @@ private void logIn(WebDriver driver) throws InterruptedException {
...
-    private boolean findFavorite(WebDriver driver, int countOfFavorites) {
-        if (driver.findElements(By.className("center full-size")).size() > countOfFavorites)
-            return true;
+    private boolean findFavorite(int countOfFavorites) throws InterruptedException {
+        Thread.sleep(5000);
+        return driver.findElements(By.className("full-size")).size() > countOfFavorites;
+    }
 
-        return false;
+    private boolean find5and90euro() throws InterruptedException{
```
Comment (HamTorosyan) [inline WebsiteGreetz/src/test/java/WebsiteTest.java:30]: - this method doesn't throw any Exception
- usage of this method is extremely narrow
- I didn't like the name 😆

File: WebsiteGreetz/src/test/java/WebsiteTest.java [hunk 2]
```diff
@@ -24,56 +22,53 @@ private void logIn(WebDriver driver) throws InterruptedException {
...
         logIn(driver);
-
-        //ավելացնենք ֆավորիտ
-
         driver.get("https://www.greetz.nl/ballonnen/denken-aan");
         Thread.sleep(3000);
-
-        WebElement item = driver.findElement(By.xpath("//a[@data-id='1142804226']/.."));
-        WebElement favoriteItem = item.findElement(By.className("b-products-grid__item-action"));
+        WebElement item = driver.findElement(By.xpath("//a[@data-id='1142804226']"));
```
Comment (HamTorosyan) [inline WebsiteGreetz/src/test/java/WebsiteTest.java:41]: can't say that name `item` is not meaningful, but I would write `targetItem`

File: WebsiteGreetz/src/test/java/WebsiteTest.java (hunk 1 above)
Comment (RipaAloyan) [reply WebsiteGreetz/src/test/java/WebsiteTest.java:30]: Yes, there was thread.sleep() at first, then I got it works without it as well and delete thread.sleep(). But forgot "throws...."

File: WebsiteGreetz/src/test/java/WebsiteTest.java [hunk 3]
```diff
@@ -24,56 +22,53 @@ private void logIn(WebDriver driver) throws InterruptedException {
...
+        WebElement favoriteItem = item.findElement(By.xpath("//parent::div//child::a[@class='b-products-grid__item-action']"));
```
Comment (HamTorosyan) [inline WebsiteGreetz/src/test/java/WebsiteTest.java:42]: actually this WebElement is not your favorite item as tells the variable name, but it is your `targetElementFavoriteButton`

File: WebsiteGreetz/src/test/java/WebsiteTest.java [hunk 4]
```diff
@@ -24,56 +22,53 @@ private void logIn(WebDriver driver) throws InterruptedException {
...
         favorite.click();
-
-        //int countOfFavorites = 0;
-        //Assert.assertTrue(findFavorite(driver, countOfFavorites));
-
-
-
+        int countOfFavorites = 0;
+        Assert.assertTrue(findFavorite(countOfFavorites));
+        //մեկ ու մեջ ա աշխատելու, մի անգամ նշումա, հաջորդ անգամը ՝ հանում
```
Comment (HamTorosyan) [inline WebsiteGreetz/src/test/java/WebsiteTest.java:50]: of course, you can and should handle that case, if the item is already in your favorites, another workflow can be, or simply you can remove it from the favorites, then add it once more. let's say you should have a fully working test method, in real life, it's not acceptable to expect TC fail in any 2nd try

File: WebsiteGreetz/src/test/java/WebsiteTest.java (hunk 1 above)
Comment (RipaAloyan) [reply WebsiteGreetz/src/test/java/WebsiteTest.java:30]: It is.I will replace the method with variable.

I will change the name.

File: WebsiteGreetz/src/test/java/WebsiteTest.java [hunk 5]
```diff
@@ -24,56 +22,53 @@ private void logIn(WebDriver driver) throws InterruptedException {
...
+        driver.quit();
     }
 
     @Test
-    public void cardPriceX3() throws InterruptedException {
-        //login
+    public void cardPriceX2() throws InterruptedException {
+        //լոգին
         System.setProperty("webdriver.chrome.driver", "src/main/resources/chromedriver.exe");
         driver = new ChromeDriver();
```
Comment (HamTorosyan) [inline WebsiteGreetz/src/test/java/WebsiteTest.java:58]: actually, you do the same steps on 36, 37 and 57, 58 lines, in codding, it's known as code repetition, and you should bypass similar cases as much as possible, I will suggest to organize your driver instantiation inside `logIn` method

File: WebsiteGreetz/src/test/java/WebsiteTest.java [hunk 6]
```diff
@@ -24,56 +22,53 @@ private void logIn(WebDriver driver) throws InterruptedException {
...
+        int countOfFavorites = 0;
+        Assert.assertTrue(findFavorite(countOfFavorites));
+        //մեկ ու մեջ ա աշխատելու, մի անգամ նշումա, հաջորդ անգամը ՝ հանում
+        driver.quit();
     }
 
     @Test
-    public void cardPriceX3() throws InterruptedException {
-        //login
+    public void cardPriceX2() throws InterruptedException {
```
Comment (HamTorosyan) [inline WebsiteGreetz/src/test/java/WebsiteTest.java:55]: let's have more descriptive and meaningfull method name

**Review by HamTorosyan — CHANGES_REQUESTED** [review]

**Review by HamTorosyan — CHANGES_REQUESTED** [review]
Let's keep some convention for branch names, If I correct it discussed in chats and/or offline lessons

File: WebsiteGreetz/src/test/java/WebsiteTest.java (hunk 2 above)
Comment (RipaAloyan) [reply WebsiteGreetz/src/test/java/WebsiteTest.java:41]: these parts are from different versions.

File: WebsiteGreetz/src/test/java/WebsiteTest.java (hunk 6 above)
Comment (RipaAloyan) [reply WebsiteGreetz/src/test/java/WebsiteTest.java:55]: these parts are from different versions.

File: WebsiteGreetz/src/test/java/WebsiteTest.java (hunk 5 above)
Comment (RipaAloyan) [reply WebsiteGreetz/src/test/java/WebsiteTest.java:58]: these parts are from different versions.

**Pull request closed without merging**